src/components/Shared/CommunityListDropdown/index.js

```diff
@@ -33,7 +34,7 @@ class CommunityListDropdown extends Component {
       <div className="CommunityListDropdown">
         {this.props.data.allCommunities.nodes.map(community => (
           <Link
-            to={`/map/community/${community.id}`}
+            to={`${(this.props.location.pathname.includes('dashboard') ? '/dashboard' : '')}/map/community/${community.id}`}
```

Comment:
re: my previous comment, this could then be`${(this.props.onDash ? '/dashboard' : '')}`